### PR TITLE
fix(dropdown): refactor vm in p-search-dropdown & p-select-dropdown

### PR DESCRIPTION
**Pull request by piggggggggy**
### To Reviewers
- [ ] Skipping review is not a problem (```style```, ```chore``` ONLY)
- [x] Not that difficult

### Type of Change
- [ ] New feature
- [ ] Bug fixes
- [ ] Feature improvement
  - [ ] These changes require a usage change
- [x] Others (performance improvement, refactoring, CI/CD, etc.)

### Checklist
- [ ] Updated Storybook documents
- [x] Tested with console

### Description
- remove `vm`, `getCurrentInstance`
- change `vm.$nextTick()` to `nextTick()`
- change `makeOptionalProxy` to `useProxyValue`

### CODE REVIEW

**Review by yuda110 — APPROVED** [review]
LGTM!

File: src/inputs/dropdown/search-dropdown/PSearchDropdown.vue [hunk 1]
```diff
@@ -194,9 +193,7 @@ export default defineComponent<SearchDropdownProps>({
             default: false,
         },
     },
-    setup(props, { emit, slots, listeners }) {
-        const vm = getCurrentInstance()?.proxy as Vue;
-
+    setup(props, { emit, slots, listeners }: SetupContext) {
```
Comment (minseolee) [inline src/inputs/dropdown/search-dropdown/PSearchDropdown.vue:196]: 👍

**Review by minseolee — APPROVED** [review]
LGTM!
I think the title of this PR's prefix should be changed to `refactor` or `fix`

**Pull request merged**